RecoTracker/MeasurementDet/plugins/TkStackMeasurementDet.cc

```diff
@@ -24,12 +33,103 @@ TkStackMeasurementDet::RecHitContainer TkStackMeasurementDet::recHits(const Traj
   HitCollectorForRecHits collector( &fastGeomDet(), theMatcher, theCPE, result );
   collectRecHits(ts, collector);
 */
+  if (data.phase2OTVectorHits().empty())
+    return result;
+  LogTrace("MeasurementTracker") << " is not empty";
+  if (!isActive(data))
+    return result;
+  LogTrace("MeasurementTracker") << " and is active";
+  /*
```

Comment:
delete commented-out code

Reply:
I'd prefer to keep this old version of the code here for now until it is agreed to go with the solution I implemented. I added an explanatory comment.